File: TrelloMini.Backend/TrelloMini.Api/Models/Card.cs
using System.ComponentModel.DataAnnotations;

namespace TrelloMini.Api.Models
{
    public class Card
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(200)]
        public string Title { get; set; } = string.Empty;
        
        [StringLength(1000)]
        public string? Description { get; set; }
        
        public int Position { get; set; }
        
        public DateTime? DueDate { get; set; }
        
        public int ListId { get; set; }
        public List List { get; set; } = null!;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: TrelloMini.Frontend/src/app/models/card.ts
export enum CardPriority {
  None = 0,
  Low = 1,
  Medium = 2,
  High = 3,
  Critical = 4
}

export interface Card {
  id: number;
  title: string;
  description?: string;
  position: number;
  dueDate?: Date;
  priority: CardPriority;
  listId: number;
  createdAt: Date;
  updatedAt: Date;
}

export interface MoveCardRequest {
  listId: number;
  position: number;
}
